python/docs/source/user_guide/sql/python_data_source.rst

```diff
@@ -59,8 +59,17 @@ Start by creating a new subclass of :class:`DataSource`. Define the source name,
         def reader(self, schema: StructType):
             return FakeDataSourceReader(schema, self.options)
 
+        def streamReader(self, schema: StructType):
+            return FakeStreamReader(schema, self.options)
 
-**Step 2: Implement the Reader**
+        def simpleStreamReader(self, schema: StructType):
```

Comment:
This could lead to confusion that they have to implement both, not either one. Could we please add comment or just duplicate the data source and describe separately?

Reply:
Comment added.